### PR TITLE
Add delete item test to verify authorization layer

### PR DESCRIPTION
**Pull request by serpient**

### CODE REVIEW

File: test/rebay_api_web/controllers/item_controller_test.exs [hunk 1]
```diff
@@ -215,6 +215,21 @@ defmodule RebayApiWeb.ItemControllerTest do
 
       assert json_response(conn, 401)["errors"] != %{}
     end
+
+    test "renders error when user is not authorized", %{conn: conn, item: item, user: user} do
+      conn = TestHelpers.valid_session(conn, user)
```
Comment (zacholauson) [inline test/rebay_api_web/controllers/item_controller_test.exs:220]: How does this `valid_session` affect this test? Both `valid_session` and the manual cookie configuration is kinda throwing me off.

File: test/rebay_api_web/controllers/item_controller_test.exs [hunk 2]
```diff
@@ -215,6 +215,21 @@ defmodule RebayApiWeb.ItemControllerTest do
...
+      |> delete(Routes.user_item_path(conn, :delete, item.user.uuid, item.uuid), [cookie: %{session_id: "test_id_token"}])
+
+      assert json_response(conn, 401)["errors"] != %{}
+    end
+
+    test "renders error when user is not the item owner", %{conn: conn, item: %Item{uuid: uuid}} do
+      unauthorized_user = TestHelpers.user_fixture()
+      conn = TestHelpers.valid_session(conn, unauthorized_user)
+      |> put(Routes.user_item_path(conn, :delete, unauthorized_user.uuid, uuid), [cookie: %{session_id: "test_id_token"}])
```
Comment (zacholauson) [inline test/rebay_api_web/controllers/item_controller_test.exs:229]: Similar comment about the manual cookie configuration here.

File: test/rebay_api_web/controllers/item_controller_test.exs (hunk 1 above)
Comment (tpetersen0308) [reply test/rebay_api_web/controllers/item_controller_test.exs:220]: This test is for the case where a user is authenticated (so there is a valid session associated with the user), but they're not authorized to delete the item they're requesting to delete. Do you think it would be sufficient to just change the test description to something like "renders error when user is authenticated but not authorized"?

File: test/rebay_api_web/controllers/item_controller_test.exs (hunk 2 above)
Comment (tpetersen0308) [reply test/rebay_api_web/controllers/item_controller_test.exs:229]: This is an edge case that only applies if the user is already logged in (`valid_session`) and the request is from an authenticated source (manually configuring the cookie in the request headers). As in my previous response, would it be sufficient to just change the test description to explicate the circumstances of the edge case?

File: test/rebay_api_web/controllers/item_controller_test.exs (hunk 1 above)
Comment (zacholauson) [reply test/rebay_api_web/controllers/item_controller_test.exs:220]: Just so I understand correctly: the item belongs to the user, but then bypasses that by providing a fake session_id on delete?

Comment (tpetersen0308) [reply test/rebay_api_web/controllers/item_controller_test.exs:220]: Yeah. I'd say invalid session_id rather than fake though, because it would apply to anything that doesn't match the id stored in the session.

Comment (zacholauson) [reply test/rebay_api_web/controllers/item_controller_test.exs:220]: I see. I was wondering if it would be more clear to create multiple users (A and B), assign the item to User A, and then try to delete with User B. Considering this is the last day of the project, I wouldn't worry too much about it, I just figured it should be called out.

Comment (tpetersen0308) [reply test/rebay_api_web/controllers/item_controller_test.exs:220]: I do have a test for that! Line 230. I wrote the first one to test the authorization that makes sure the user making the request is the same user in the session, and then wrote that authorization, then realized I needed an authorization checking if the requester is the item owner, and wrote the other test. But now that I think of it, the latter may have rendered the former irrelevant.
